occurrence-ws/src/main/java/org/gbif/occurrence/ws/provider/hive/ConnectionPool.java

```diff
@@ -0,0 +1,136 @@
+package org.gbif.occurrence.ws.provider.hive;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.apache.nifi.attribute.expression.language.StandardPropertyValue;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.PropertyValue;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.dbcp.hive.HiveConnectionPool;
+import org.apache.nifi.mock.MockControllerServiceInitializationContext;
+import org.apache.nifi.reporting.InitializationException;
+import org.gbif.utils.file.properties.PropertiesUtil;
+import org.gbif.ws.app.ConfUtils;
+
+/**
+ * 
+ * ConnectionPool for Hive JDBC connection.
+ *
+ */
+public class ConnectionPool{
```

Comment:
This class is being used incorrectly, a connection pool instance is created each time a validation is done, and instance should be  kept at the ws level or at the guice model